semiphemeral/twitter.py

```diff
@@ -316,3 +324,34 @@ def delete(self):
                     self.common.session.commit()
 
             self.common.session.commit()
+
+    def import_dump(self, filepath):
+        if self.common.settings.get('delete_tweets'):
+            with open(os.path.join(filepath, 'tweet.js'), 'r') as f:
+                # Skip the JS variable assignment at the start of this file
+                f.read(25)
+                tweets = json.load(f)
+            click.echo('Importing {} tweets from {}'.format(len(tweets), filepath))
+
+            current_user = self.api.me()
+
+            def parse_tweet(tweet):
+                """ Parse a JSON tweet into a tweepy object and insert missing author. """
+                t = Status.parse(self.api, tweet)
+                t.author = current_user
```

Comment:
> The archive I just downloaded from Twitter does not appear to flag retweets as such - they just appear (and get inserted into the DB) as normal tweets. I can't work out if this is a Twitter bug or not. I'm not too concerned about this as I want to delete retweets as well.

I'm running this importer right now (thanks a ton! WAY faster than doing a decade of tweets via the api!), and it appears to be flagging the tweet authors correctly (it's printing the RT'd tweet's author to the console). I haven't dived in too deeply, but this line may be the culprit on storing RTs as normal tweets?